Podcast/SearchDiscoverViewController.swift

```diff
@@ -165,36 +165,39 @@ class SearchViewController: ViewController, UISearchControllerDelegate, UITableV
         view.addSubview(pastSearchesTableView)
         mainScrollView = pastSearchesTableView
 
-        tabUnderlineView = UnderlineTabBarView(frame: CGRect(x: 0, y: 0, width: view.frame.width, height: tabBarUnderlineViewHeight))
-        tabUnderlineView.setUp(sections: SearchType.allValues.map{ type in type.toString() })
-        tabUnderlineView.delegate = self
-        view.addSubview(tabUnderlineView)
-        tabUnderlineView.isHidden = true
-
         searchITunesHeaderView = SearchHeaderView(frame: .zero, type: .itunes)
         searchITunesHeaderView?.delegate = self
         addSearchITunesHeader()
 
         pastSearchesTableView.snp.makeConstraints { make in
             make.edges.equalToSuperview()
         }
-
-        tabUnderlineView.snp.makeConstraints { make in
-            make.leading.trailing.width.equalToSuperview()
-            make.top.equalTo(view.safeAreaLayoutGuide.snp.top)
-            make.height.equalTo(tabBarUnderlineViewHeight)
+        
+        searchResultsTableView.snp.makeConstraints { make in
+            make.top.equalTo(view.safeAreaLayoutGuide)
+            make.leading.trailing.bottom.equalToSuperview()
+        }
+        
+        discoverContainerView = UIView()
+        view.addSubview(discoverContainerView)
+        
+        discoverVC = DiscoverViewController()
+        addChildViewController(discoverVC)
+        discoverContainerView.addSubview(discoverVC.view)
+        
+        discoverContainerView.snp.makeConstraints { (make) in
+            make.edges.equalToSuperview()
         }
 
-        searchResultsTableView.snp.makeConstraints { make in
-            make.top.equalTo(tabUnderlineView.snp.bottom)
-            make.leading.trailing.width.bottom.equalToSuperview()
+        discoverVC.view.snp.makeConstraints { (make) in
+            make.edges.equalToSuperview()
         }
 
         hasLoaded = true
     }
 
     func addSearchITunesHeader() {
-        guard !didDismissItunesHeaderForQuery && tabUnderlineView.selectedIndex == 0 else {
+        guard !didDismissItunesHeaderForQuery else {
             searchResultsTableView.tableHeaderView = nil
             return
```

Comment:
you might need to call `layoutIfNeeded` again here to get rid of constraint errors when dismissing

Reply:
let's remove this itunes header .. see my comment in `SearchHeaderView`